From bd704c06188608aa139eb805dc04a435209d6e4d Mon Sep 17 00:00:00 2001
Message-Id: <bd704c06188608aa139eb805dc04a435209d6e4d.1423172493.git.feitong.yi@intel.com>
In-Reply-To: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
References: <18bb7324f916bffbc75e07ea0ffd3e6e68e7b6d8.1423172493.git.feitong.yi@intel.com>
From: Gaurav K Singh <gaurav.k.singh@intel.com>
Date: Thu, 15 Jan 2015 01:11:04 +0530
Subject: [PATCH 26/29] MUST_REBASE [VPG]: video/adf/intel: Use DSI Pll1 for
 enabling MIPI DSI on Port C

DSI Pll1 is used for enabling DSI on Port C.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-4868
Change-Id: Iad6c1611b33b3370f8f57b21c594b81115b70dc8
Signed-off-by: Gaurav K Singh <gaurav.k.singh@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_dsi_pll.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_dsi_pll.c b/drivers/video/adf/intel/core/vlv/vlv_dsi_pll.c
index a3dc575..0e901ae 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dsi_pll.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dsi_pll.c
@@ -230,7 +230,10 @@ static void vlv_dsi_pll_configure(struct vlv_pll *pll)
 		return;
 	}
 
-	dsi_mnp.dsi_pll_ctrl |= DSI_PLL_CLK_GATE_DSI0_DSIPLL;
+	if (pll->port_id == PORT_A)
+		dsi_mnp.dsi_pll_ctrl |= DSI_PLL_CLK_GATE_DSI0_DSIPLL;
+	else
+		dsi_mnp.dsi_pll_ctrl |= DSI_PLL_CLK_GATE_DSI1_DSIPLL;
 
 	pr_info("dsi pll div %08x, ctrl %08x\n",
 			dsi_mnp.dsi_pll_div, dsi_mnp.dsi_pll_ctrl);
-- 
1.7.9.5

